### PR TITLE
Add Launcher.TryOpenAsync

### PR DESCRIPTION
**Pull request by Redth**
### Description of Change ###

Added:
```csharp
Task<bool> Launcher.TryOpenAsync(Uri);
Task<bool> Launcher.TryOpenAsync(string);
```

### Bugs Fixed ###

- Related to discussion in Xamarin.Forms https://github.com/xamarin/Xamarin.Forms/pull/6623

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

File: Xamarin.Essentials/Launcher/Launcher.android.cs [hunk 1]
```diff
@@ -45,5 +45,15 @@ static Task PlatformOpenAsync(OpenFileRequest request)
 
             return Task.CompletedTask;
         }
+
+        static async Task<bool> PlatformTryOpenAsync(Uri uri)
+        {
+            var canOpen = await PlatformCanOpenAsync(uri);
```
Comment (StephaneDelcroix) [inline Xamarin.Essentials/Launcher/Launcher.android.cs]:
```suggestion
            var canOpen = await PlatformCanOpenAsync(uri).ConfigureAwait(false);
```

File: Xamarin.Essentials/Launcher/Launcher.android.cs [hunk 2]
```diff
@@ -45,5 +45,15 @@ static Task PlatformOpenAsync(OpenFileRequest request)
...
+
+            if (canOpen)
+                await PlatformOpenAsync(uri);
```
Comment (StephaneDelcroix) [inline Xamarin.Essentials/Launcher/Launcher.android.cs]:
```suggestion
                await PlatformOpenAsync(uri).ConfigureAwait(false);
```

File: Xamarin.Essentials/Launcher/Launcher.tizen.cs [hunk 3]
```diff
@@ -54,5 +54,15 @@ static Task PlatformOpenAsync(OpenFileRequest request)
 
             return Task.CompletedTask;
         }
+
+        static async Task<bool> PlatformTryOpenAsync(Uri uri)
+        {
+            var canOpen = await PlatformCanOpenAsync(uri);
```
Comment (StephaneDelcroix) [inline Xamarin.Essentials/Launcher/Launcher.tizen.cs]:
```suggestion
            var canOpen = await PlatformCanOpenAsync(uri).ConfigureAwait(false);
```

File: Xamarin.Essentials/Launcher/Launcher.tizen.cs [hunk 4]
```diff
@@ -54,5 +54,15 @@ static Task PlatformOpenAsync(OpenFileRequest request)
...
+
+            if (canOpen)
+                await PlatformOpenAsync(uri);
```
Comment (StephaneDelcroix) [inline Xamarin.Essentials/Launcher/Launcher.tizen.cs]:
```suggestion
                await PlatformOpenAsync(uri).ConfigureAwait(false);
```

File: Xamarin.Essentials/Launcher/Launcher.uwp.cs [hunk 5]
```diff
@@ -24,5 +24,15 @@ static async Task PlatformOpenAsync(OpenFileRequest request)
 
             await WinLauncher.LaunchFileAsync(storageFile).AsTask();
         }
+
+        static async Task<bool> PlatformTryOpenAsync(Uri uri)
+        {
+            var canOpen = await PlatformCanOpenAsync(uri);
```
Comment (StephaneDelcroix) [inline Xamarin.Essentials/Launcher/Launcher.uwp.cs]:
```suggestion
            var canOpen = await PlatformCanOpenAsync(uri).ConfigureAwait(false);
```

File: Xamarin.Essentials/Launcher/Launcher.uwp.cs [hunk 6]
```diff
@@ -24,5 +24,15 @@ static async Task PlatformOpenAsync(OpenFileRequest request)
...
+
+            if (canOpen)
+                return await WinLauncher.LaunchUriAsync(uri).AsTask();
```
Comment (StephaneDelcroix) [inline Xamarin.Essentials/Launcher/Launcher.uwp.cs]:
```suggestion
                return await WinLauncher.LaunchUriAsync(uri).AsTask().ConfigureAwait(false);
```

**Comment by jamesmontemagno** [comment]
@Redth i think all we need are docs here and good to go

**Comment by Redth** [comment]
Yep @newky2k is going to do the rest of this puppy.

File: docs/en/Xamarin.Essentials/Launcher.xml [hunk 7]
```diff
@@ -133,5 +133,49 @@
...
+        <AssemblyVersion>1.0.0.0</AssemblyVersion>
+      </AssemblyInfo>
+      <ReturnValue>
+        <ReturnType>System.Threading.Tasks.Task&lt;System.Boolean&gt;</ReturnType>
+      </ReturnValue>
+      <Parameters>
+        <Parameter Name="uri" Type="System.String" />
+      </Parameters>
+      <Docs>
+        <param name="uri">To be added.</param>
```
Comment (jamesmontemagno) [inline docs/en/Xamarin.Essentials/Launcher.xml]: We should fill in these docs. Do you have DocWriter installed?

Comment (newky2k) [reply docs/en/Xamarin.Essentials/Launcher.xml]: Could we populate these with inline comments?  i'll install docwriter

**Pull request merged**